### ISSUE TITLE
Update multiple Group Chat dialogs

### ISSUE THREAD

**Issue by unoiatech** [QuickBlox/quickblox-ios-sdk#891]
**New Issue Checklist**

 - [x] I have read the [Documentation](https://quickblox.com/developers/IOS)
 - [x] I have searched for a similar issue in the [project](https://github.com/QuickBlox/quickblox-ios-sdk/issues) and found none

**Environment details**

Info                           | Value                                       |
-------------------------------|---------------------------------------------|
iOS Version                    | e.g. 11                                    |                                          
Quickblox iOS SDK version      | e.g. 2.9.2                                  |
Xcode Version                  | e.g. Xcode 9                              |
Repro with our demo sample     | e.g. does it happen with our demo sample?   |

**Did this work before?**
No

**Expected behavior**
Just like QBCOCustomObjects

 ```
+ (QBRequest *)updateObjects:(NSArray<QBCOCustomObject *> *)objects
                   className:(NSString *)className
                successBlock:(nullable void (^)(QBResponse *response, NSArray<QBCOCustomObject *> * _Nullable objects, NSArray<NSString *> * _Nullable notFoundObjectsIds))successBlock
                  errorBlock:(nullable QBRequestErrorBlock)errorBlock;
```
I want to update multiple chat dialogs with One Method. Currently there's a method that updates single Chat Dialog.
**Actual behavior**
 Currently there's a method that updates single Chat Dialog.
`QBRequest updateDialog:`

**Comment by Pro100Andrey** [comment]
Hi @unoiatech,

Could you expand a little bit on why you need `update multiple chat dialogs with One Method` ?

Thanks.

**Comment by unoiatech** [comment]
Thanks for the reply @Pro100Andrey ,

I have some functionality in my app where I need to update the Dialog Data for multiple Chat Dialogs.

**Comment by Pro100Andrey** [comment]
@unoiatech We have 3 fields for update chat dialog - `occupants_ids`, `name` and `photo`:

Fields to update:

Field | Available operators | Description | Usage example
-- | -- | -- | --
occupants_ids | pull_all,push_all | Update dialog occupants.Use push_all operator to add new occupants.Use pull_all to remove occupants | push_all[occupants_ids][]=55,56pull_all[occupants_ids][]=22,23
name |   | Name of the dialog | name=Chat with Garry and John
photo |   | Photo of the dialog | photo=674

Could you please provide your scenario (update occupants_ids  or something else) for update multiple Chat Dialogs ?

**Comment by unoiatech** [comment]
Here is my Code to update Single Chat Dialog:

```
                       NSDictionary *data;
                        
                        data = @{@"class_name": @"Dialog_Data", @"seller_id":[NSNumber numberWithInteger:GLB.chatProfileDetailUserID], @"buyer_id": [NSNumber numberWithInteger:LoginUserId], @"job_id": [NSString stringWithFormat:@"%d",GLB.jobID], @"job_status": @"completed"};
                        
                        singleDialog.data = data;
                        NSLog(@"single dialog id : %@",singleDialog);
                        
                        [QBRequest updateDialog:singleDialog successBlock:^(QBResponse * _Nonnull response, QBChatDialog * _Nullable chatDialog) {
                            NSLog(@"updated dialog : %@",singleDialog);
                        } errorBlock:^(QBResponse * _Nonnull response) {
                            NSLog(@"error updating Dialog %@",response.error.error);
                        }];
```

In this scenario,  I am updating "job_status" from "Pending" to "Completed". In some cases, I have to update same for multiple dialogs at one time.

**Comment by Pro100Andrey** [comment]
@unoiatech,
How many chat dialogs you want to update in single query? If it only a few then you can run multiple request in parallel and achieve the same behaviour.

Thanks.

**Comment by Pro100Andrey** [comment]
@unoiatech,

Also you can use another approach using custom objects. In custom parameters of the chat dialog instance you can set the ID of the created custom object and use method of custom object updates

**Comment by Pro100Andrey** [comment]
@unoiatech, Any updates?

Thanks.

**Comment by unoiatech** [comment]
Dialogs count may be around 40-50.. 
I tried with Custom Objects but could not achieve the results. So, Dialogs worked for me. Only issue I am facing with dialogs is Updating Multiple.

**Comment by VitaliyGuru** [comment]
@unoiatech, 

> I tried with Custom Objects but could not achieve the results
Could you give an example of this?

**Comment by unoiatech** [comment]
In my app, I have a Job. Multiple Users can Apply on that job. Each user will have individual Chat dialog associated to the Job. Means, if 30 users apply on a job, there will 30 Chat dialogs. 

Now, if the Owner of job accept one of the Proposal from 30. The accepted one will be marked as accept and others will be marked as Rejected.

We are keeping the Job_status with each dialog using Dialog_Data. Which we need to update when the above step happens.

That is why I want to update Multiple chat dialogs at once.

**Comment by VitaliyGuru** [comment]
@unoiatech, For your scenario there is no need to bind QBChatDialogs to the custom object.


|Job (Custom object )|
|---------------------|
|User_ids[Array]      |
|owner_ID[Integer]                    |
|Approved User ID[Integer]  |
|..etc |

As owner:
1. You can retrieve all custom objects where 'owner_ID == Your_ID'.
2. You can retrieve all private dialogs from rest(or cache) using 'User ids'.
3. You can get the status of each dialog with  'Approved User ID'.

As user:
1. You can get all jobs associated with you using the request like 'Your_ID IN 'Job'.User_ids '
2. You can get the status of the job with  'Approved User ID'.

When the owner accepts the proposal he can send the notification message with some info(depends on you which exactly) for all associated dialogs to notify the users.

**Comment by Pro100Andrey** [comment]
Hi @unoiatech, any updates ?
